marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/reader/HubDataSourceReader.java

```diff
@@ -153,6 +161,10 @@ private InputStreamHandle readCustomInitializeApiDefinition(Map<String, String>
                 throw new RuntimeException("Unable to read custom API module for initializing Read job: " + options.get(key) + "; cause: " + ex.getMessage(), ex);
             }
         }
+        else {
+            loadModuleIfNotPresent("/marklogic-data-hub-spark-connector/initializeRead.sjs", Format.TEXT);
```

Comment:
I believe this needs to load readLib.sjs and partition-lib.xqy as well, since initializeRead.sjs depends on those.

Reply:
Hi @rjrudin, readLib.sjs and partition-lib.xqy are loaded to the data-hub-MODULES in the "determineReadRowsEndpointParams" method so they will be present in the database before we reach here. Also, currently i have not put any condition for loading  readLib.sjs and partition-lib.xqy. Hence they will always be loaded if they are not present. Please let me know if we need to add a condition.

Reply:
I see - determineReadRowsEndpointParams is called before initializeRead is called. Sounds good.